UI/CUIControls.cpp

```diff
@@ -1924,3 +1928,117 @@ void RotatingGraphic::Render() {
+    m_unchecked_color(unchecked_clr),
+    m_checked_color(checked_clr)
+{}
+
+void CUIToggleRepresenter::RenderUnchecked(const CUIToggleButton& button) const {
+    GG::Clr icon_clr(m_unchecked_color);
+    GG::Clr bg_clr(ClientUI::CtrlColor());
+    GG::Clr border_clr(button.Disabled() ? DisabledColor(ClientUI::CtrlBorderColor()) : ClientUI::CtrlBorderColor());
+
+    const int BORDER_THICKNESS = 1;
+    const int ICON_SIZE_REDUCE = BORDER_THICKNESS * 2;
+
+    GG::Pt icon_ul = button.UpperLeft();
+    GG::Pt icon_lr = button.LowerRight();
+    GG::Pt border_ul = icon_ul;
+    GG::Pt border_lr = icon_lr;
+
+    // tone color for disabled button or single icon with same color
+    if (button.Disabled() || ((m_checked_color == m_unchecked_color) && (m_checked_icon == m_unchecked_icon)))
+        icon_clr = DisabledColor(icon_clr) * 0.7;
+
+    // highlight on mouseover
+    if (button.IsMouseover()) {
+        AdjustBrightness(border_clr, 100);
+        AdjustBrightness(icon_clr, 1.7);
+    }
+
+    // shrink icon for border
+    icon_ul.x += ICON_SIZE_REDUCE;
+    icon_ul.y += ICON_SIZE_REDUCE;
+    icon_lr.x -= ICON_SIZE_REDUCE;
+    icon_lr.y -= ICON_SIZE_REDUCE;
+
+    // render border
+    AngledCornerRectangle(border_ul, border_lr, bg_clr, border_clr, CUIBUTTON_ANGLE_OFFSET, BORDER_THICKNESS);
+
+    // render icon
+    glColor(icon_clr);
+    m_unchecked_icon->OrthoBlit(icon_ul, icon_lr);
+}
+
+void CUIToggleRepresenter::RenderChecked(const CUIToggleButton& button) const {
+    GG::Clr icon_clr(!button.Disabled() ? m_checked_color : DisabledColor(m_checked_color) * 0.7);
+    GG::Clr bg_clr(ClientUI::CtrlColor());
+    GG::Clr border_clr(button.Disabled() ? DisabledColor(ClientUI::CtrlBorderColor()) : ClientUI::CtrlBorderColor());
+
+    const int BORDER_THICKNESS = 1;
+    const int ICON_SIZE_REDUCE = BORDER_THICKNESS * 2;
+
+    GG::Pt icon_ul = button.UpperLeft();
+    GG::Pt icon_lr = button.LowerRight();
+    GG::Pt border_ul = icon_ul;
+    GG::Pt border_lr = icon_lr;
+
+    // highlight on mouseover
+    if (button.IsMouseover()) {
+        AdjustBrightness(border_clr, 100);
+        AdjustBrightness(icon_clr, 1.7);
+    }
+
+    // shrink icon for border
+    icon_ul.x += ICON_SIZE_REDUCE;
+    icon_ul.y += ICON_SIZE_REDUCE;
+    icon_lr.x -= ICON_SIZE_REDUCE;
+    icon_lr.y -= ICON_SIZE_REDUCE;
+
+    // render border
+    AngledCornerRectangle(border_ul, border_lr, bg_clr, border_clr, CUIBUTTON_ANGLE_OFFSET, BORDER_THICKNESS);
+
+    // render icon
+    glColor(icon_clr);
+    m_checked_icon->OrthoBlit(icon_ul, icon_lr);
+}
+
+CUIToggleButton::CUIToggleButton(const std::string& tooltip, boost::shared_ptr<CUIToggleRepresenter> representer) :
+    StateButton("", ClientUI::GetFont(), GG::FORMAT_NONE, GG::CLR_ZERO, representer),
+    m_representer(representer)
+{
+    SetBrowseInfoWnd(boost::shared_ptr<GG::BrowseInfoWnd>(new TextBrowseWnd(tooltip, "")));
+    GG::Connect(CheckedSignal, &PlayButtonCheckedSound, -1);
```

Comment:
Use the `void CUIToggleButtonRepresenter::OnChecked(bool checked) const` method instead to call PlayButtonCheck.  See `void CUICheckBoxRepresenter::OnChecked(bool checked) const` (UI/CUIControls.cpp:363) as an example.